rclcpp/test/rclcpp/test_intra_process_manager_with_allocators.cpp

```diff
@@ -57,7 +58,10 @@ struct MyAllocator
       return nullptr;
     }
     num_allocs++;
-    return static_cast<T *>(std::malloc(size * sizeof(T)));
+    // Use sizeof(char) in place for sizeof(void)
+    constexpr size_t value_size = sizeof(
+      typename std::conditional<!std::is_void<T>::value, T, char>::type);
```

Comment:
I guess this is fine. I spent some time looking at libcxx's code and I couldn't find anything in the `void` specialization that avoids `sizeof(void)`, but I think that's technically a GCC extension. I wondered if `uint8_t` is better than `char` here (even though they are functionally the same), and I was actually looking for what they did when I couldn't find logic similar to this in their `allocate` function for `std::allocator`.

Reply:
I was actually hoping to find something on the `std::allocator` class or the `std::allocator_traits` that we could use instead, e.g. something like `std::allocator<T>::element_size`, but it doesn't exist.

Reply:
Hmm, the `tlsf_cpp` allocator in `realtime_support` is showing the same issue https://ci.ros2.org/view/nightly/job/nightly_linux_clang_libcxx/778/console#console-section-24.

Reply:
There is also a new warning in all builds, likely related to this: https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_debug/1593/gcc/new/

Reply:
Same root cause.